### PR TITLE
Document ReferencePosition

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Documents common `ReferencePosition` methods.

### CODE REVIEW

File: packages/dds/merge-tree/src/referencePositions.ts [hunk 1]
```diff
@@ -56,12 +56,35 @@ export function refHasRangeLabels(refPos: ReferencePosition): boolean {
...
     refType: ReferenceType;
 
+    /**
+     * Gets the segment that this reference position is semantically associated with. Returns undefined iff the
+     * reference became detached from the string.
+     */
     getSegment(): ISegment | undefined;
+
+    /**
+     * Gets the offset for this reference position relative to the segment it's associated to.
```
Comment (jzaffiro) [inline packages/dds/merge-tree/src/referencePositions.ts:78]: This is slightly confusing to me - is it the offset from the root relative to this reference position's segment? I may also just not have a great understanding of offset, so feel free to disregard this comment if that is the case.

**Review by jzaffiro — APPROVED** [review]
Looks good for the most part, I think the wording of the getOffset() description is slightly confusing though.

File: packages/dds/merge-tree/src/referencePositions.ts (hunk 1 above)
Comment (Abe27342) [reply packages/dds/merge-tree/src/referencePositions.ts]: nah, it's good feedback. I was motivated to doc this interface after getting a question from someone trying to use it. We want it to be abundantly clear that this is *not* an absolute position within the merge-tree. I added an example to demonstrate that.

File: packages/dds/merge-tree/src/referencePositions.ts [hunk 2]
```diff
@@ -56,12 +56,38 @@ export function refHasRangeLabels(refPos: ReferencePosition): boolean {
...
 
+    /**
+     * Gets the segment that this reference position is semantically associated with. Returns undefined iff the
+     * reference became detached from the string.
+     */
     getSegment(): ISegment | undefined;
+
+    /**
+     * Gets the offset for this reference position within its associated segment.
+     * @example
```
Comment (jzaffiro) [inline packages/dds/merge-tree/src/referencePositions.ts:79]: This is super helpful!

**Review by jzaffiro — APPROVED** [review]

**Pull request merged**